File: Reporting/VB/ReportMainDemo/BaseForms/RibbonReportDesigner/MainFormBase.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraReports.UserDesigner

Namespace RibbonReportDesigner

    Public Class MainFormBase
        Inherits RibbonForm

        Public Sub New()
            Text = "Report Designer"
#If DEBUG
            Text += $" v{AssemblyInfo.VersionShort}"
#End If
        End Sub

        Public Overridable Sub OpenReport(ByVal newReport As DevExpress.XtraReports.UI.XtraReport)
        End Sub

        Public Overridable ReadOnly Property ActiveXRDesignPanel As XRDesignPanel
            Get
                Return Nothing
            End Get
        End Property
    End Class
End Namespace

File: WinForms/DevExpress.VideoRent.Win/CS/DevExpress.VideoRent.Win/ModulesReports/RibbonReportDesigner/MainFormBase.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraReports.UserDesigner;

namespace RibbonReportDesigner {
    public class MainFormBase : RibbonForm {
        public MainFormBase() {
            Text = "Report Designer";
#if DEBUG
            Text += $" v{AssemblyInfo.VersionShort}";
#endif
        }
        public virtual void OpenReport(DevExpress.XtraReports.UI.XtraReport newReport) {
        }
        public virtual XRDesignPanel ActiveXRDesignPanel { get { return null; } }
    }
}
